armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -623,20 +575,32 @@ def mapHotTempToComponents(self, tempGrid, tempField):
 
             blockAveTemp = mean(tmpMapping)
             for c in b:
-                self._oldHotTemp[c] = c.temperatureInC  # stash the "old" hot temp
-                # set component volume to be evaluated at "old" hot temp
-                c.p.volume = c.getArea(cold=self._oldHotTemp[c]) * c.parent.getHeight()
+                self.componentReferenceHeight[c] = b.getHeight()
+                # store the temperature from previous state (i.e., reference temp)
+                self.componentReferenceTemperature[c] = c.temperatureInC
+                # set component volume to be evaluated at reference temp
+                c.p.volume = (
+                    c.getArea(cold=self.componentReferenceTemperature[c])
+                    * c.parent.getHeight()
+                )
                 # DO NOT use self.setTemperature(). This calls changeNDensByFactor(f)
-                # and ruins mass conservation via number densities. Instead,
-                # set manually.
+                # and ruins mass conservation via number densities. Instead, set manually.
                 c.temperatureInC = blockAveTemp
 
     def computeThermalExpansionFactors(self):
         """computes expansion factors for all components via thermal expansion"""
 
         for b in self._a:
             for c in b:
```

Comment:
So, what happens if a component/block get removed from the Reactor?

For instance, if an entire assembly is moved to the Spent Fuel Pool (SFP).

Will further calls to this code still work?  That is, will we get wrong data returned?  If someone adds a NEW assembly to the Reactor to replace the one in the SFP, will we see that new Assembly here?

Reply:
I wouldn't expect it to be a problem. The axial expansion changer is designed to live for a short period of time - only as long as it's needed. For instance, in `Core::processLoading` it goes out of scope at the conclusion of the method and is released from memory. 

So at BOL during `Core::processLoading` we fire up an instance of the `AxialExpansionChanger`, loop over each assembly in the core, assign an assembly to the instance of the changer, expand it, and then move on to the next. Each time an assembly is set to the changer, all of the data on the changer is reset for that specific assembly.

The `AxialExpansionChanger` is designed to be blind to swapping of assemblies (e.g., shuffling). So if were to do some shuffle at some arbitrary point during a run, we could follow the same process to loop over the core and expand each assembly in the Core.

Reply:
Okay, I see that now:

https://github.com/terrapower/armi/blob/c2d6c767ef0f5e95c76d9d3b21b03eab1d7d4636/armi/reactor/reactors.py#L2243-L2248

So, I don't have to worry about persistence.

Thanks!